### PR TITLE
🐞 fix(setup):

### PR DESCRIPTION
**Pull request by Jesse3692**
Fixed a specific bug in Windows 10: UnicodeDecodeError

When I was building the application, I encountered the following error:

```python
  Traceback (most recent call last):
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 349, in <module>
      main()
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "d:\github\yifeng-chia\venv\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 117, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 154, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "C:\Users\Administrator\AppData\Local\Temp\pip-build-env-zkqkhqel\overlay\Lib\site-packages\setuptools\build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 112, in <module>
      long_description=open("README.md").read(),
  UnicodeDecodeError: 'gbk' codec can't decode byte 0x9a in position 412: illegal multibyte sequence
```

### CODE REVIEW

**Review by Jesse3692 — COMMENTED** [review]
this pull can resolve UnicodeDecodeError

**Comment by arvidn** [comment]
I'm a bit surprised this is an issue in python, but I would think we would want to set the global locale for the process early on, to something that specifies utf-8 as the encoding. I defer to a python expert.

**Comment by aqk** [comment]
Thank you for this fix.

**Review by aqk — DISMISSED** [review]

**Review by Jesse3692 — COMMENTED** [review]
use the same quote style

**Review by Jesse3692 — COMMENTED** [review]
use 'with' for resource-allocating operations, automatically close file descriptors to release system resources

**Review by Jesse3692 — COMMENTED** [review]
fix a syntax error

**Review by Jesse3692 — COMMENTED** [review]
fix a syntax error

**Review by arvidn — DISMISSED** [review]

**Comment by hoffmang9** [comment]
To fix your lint issue do the following inside the venv.

```
pip install black flake8
black setup.py
flake8 setup.py
```
And check in those changes.

**Review by Jesse3692 — COMMENTED** [review]
Some code lengths have been changed to solve the length problem.

File: setup.py [hunk 1]
```diff
@@ -14,12 +14,15 @@
...
-    #  See https://github.com/frispete/keyrings.cryptfile/issues/15
+    # Python cryptography library for TLS - keyring conflict
+    "cryptography==3.4.7",
+    # Store keys in MacOS Keychain, Windows Credential Locker
+    "keyring==23.0.1",
+    # Secure storage for keys on Linux (Will be replaced)
+    "keyrings.cryptfile==1.3.4",
+    # "keyrings.cryptfile==1.3.8"
+    # Secure storage for keys on Linux (Will be replaced)
+    # See https://github.com/frispete/keyrings.cryptfile/issues/15
```
Comment (arvidn) [inline setup.py:25]: where these changes deliberate or a rebase-issue?

Comment (Jesse3692) [reply setup.py:25]: Sorry, my original intention was just to modify the coding error problem, and then found that the reading of the file is also a bit of a problem, so I added a function containing the with to automatically manage the file. But I found that previous submissions had not passed, and one contributor told me that I needed to use black and flake8 to check the file format and code style, so I modified the previous comments.

**Pull request closed without merging**